src/routes/(index)/reset-password.tsx

```diff
@@ -0,0 +1,129 @@
+import React, { useState } from "react";
+import CanvasCard from "@/components/ui/CanvasCard";
+import { DefaultButton } from "@/components/ui/defaultButton";
+import { createFileRoute } from "@tanstack/react-router";
+import { Input } from "@/components/ui/input";
+import { Typography } from "@/components/ui/typography";
+
+export const Route = createFileRoute("/(index)/reset-password")({
+  component: ResetPasswordPage,
+});
+
+function validatePassword(password: string) {
+  // Pelo menos uma letra maiúscula e um número
+  return /[A-Z]/.test(password) && /\d/.test(password);
+}
+
+export default function ResetPasswordPage() {
+  const [password, setPassword] = useState("");
+  const [confirm, setConfirm] = useState("");
+  const [error, setError] = useState("");
+  const [success, setSuccess] = useState("");
+  const [loading, setLoading] = useState(false);
+
+  const handleSubmit = async (e: React.FormEvent) => {
+    e.preventDefault();
+    setError("");
+    setSuccess("");
+
+    if (!password || !confirm) {
+      setError("Preencha todos os campos.");
+      return;
+    }
+    if (!validatePassword(password)) {
+      setError("A senha deve conter ao menos uma letra maiúscula e um número.");
+      return;
+    }
+    if (password !== confirm) {
+      setError("As senhas devem ser iguais.");
+      return;
+    }
+
+    setLoading(true);
+    try {
+      const response = await fetch("/api/reset-password", {
```

Comment:
```suggestion
      const response = await fetch("/api/reset-password", {
```
Mesma coisa aqui, faz em um arquivo separado. Olha na outra branch que eu citei como exemplo